src/vs/workbench/contrib/workspace/browser/workspaceTrustEditor.ts

```diff
@@ -254,26 +268,33 @@ export class WorkspaceTrustEditor extends EditorPane {
 		this.affectedFeaturesContainer = append(parent, $('.workspace-trust-features'));
 	}
 
-	private renderAffectedFeatures(numExtensions: number): void {
+	private renderAffectedFeatures(numSettings: number, numExtensions: number): void {
 		clearNode(this.affectedFeaturesContainer);
 		const trustedContainer = append(this.affectedFeaturesContainer, $('.workspace-trust-limitations.trusted'));
 		this.renderLimitationsHeaderElement(trustedContainer, localize('trustedWorkspace', "Trusted Workspace"), this.getHeaderTitleIconClassNames(WorkspaceTrustState.Trusted));
-		this.renderLimitationsListElement(trustedContainer, [
+
+		const limitationsInTrustedWorkspace = [
 			localize('trustedTasks', "Tasks will be allowed to run"),
 			localize('trustedDebugging', "Debugging will be enabled"),
-			localize('trustedSettings', "All workspace settings will be applied"),
-			localize('trustedExtensions', "All extensions will be enabled"),
-		], checkListIcon.classNamesArray);
+		];
+		if (numSettings > 0) {
+			limitationsInTrustedWorkspace.push(localize('trustedSettings', "All workspace settings will be applied"));
+		}
+		limitationsInTrustedWorkspace.push(localize('trustedExtensions', "All extensions will be enabled"));
+		this.renderLimitationsListElement(trustedContainer, limitationsInTrustedWorkspace, checkListIcon.classNamesArray);
 
 		const untrustedContainer = append(this.affectedFeaturesContainer, $('.workspace-trust-limitations.untrusted'));
 		this.renderLimitationsHeaderElement(untrustedContainer, localize('untrustedWorkspace', "Untrusted Workspace"), this.getHeaderTitleIconClassNames(WorkspaceTrustState.Untrusted));
 
-		this.renderLimitationsListElement(untrustedContainer, [
+		const limitationsInUntrustedWorkspace = [
 			localize('untrustedTasks', "Tasks will be disabled"),
 			localize('untrustedDebugging', "Debugging will be disabled"),
-			localize('untrustedSettings', "Unsafe workspace settings will not be applied"),
-			localize('untrustedExtensions', "[{0} extensions](command:{1}) will be disabled or limit functionality", numExtensions, 'workbench.extensions.action.listTrustRequiredExtensions')
-		], xListIcon.classNamesArray);
+		];
+		if (numSettings > 0) {
```

Comment:
I like what you are trying to do here, but I wonder if we should still include an entry even if no workspace settings are present. This way the user knows that if they add workspace settings somehow, that it is a limitation. We could still have special wording like `Unsafe worksace settings are not applied. No such settings are currently defined`

Reply:
Makes sense. Added following general message in case there are no settings requiring trust.

**Workspace settings requiring trust will not be applied.**